### DISCUSSION TITLE
my app works fine locally but fails with test

### DISCUSSION THREAD

**Discussion by Prashant-Atla** [DevSkillsHQ/discusssions#31] [Help] [answered]
Hi Anton,
My app works fine locally but when I commit to the implementation branch it fails in the test env 
I get the error below, I dont understand why can you use out with that ?

#error 
start your docker-compose envirnoment here
http://localhost:5000/ping timed out on retry 91 of 3, elapsed 90266ms, limit 90000ms
Error: connect ECONNREFUSED 127.0.0.1:5000

**Comment by fenske** [comment, marked as answer]
Hi there. Your `build` and `start` scripts should respectively build and start your application as those commands are used in the pipeline to spin up your app. So as soon as you initialize them, the app will be spun up, and the tests will run against it.

**Reply by Prashant-Atla** [reply]
In my application and in point 3 its mentioned to spin up application using docker compose, so I wouldn't need to build except to run docker-compose?

**Reply by fenske** [reply]
If you believe that build is unnecessary in your case, then yeah, start should be enough.
